### PR TITLE
Update fluentd es

### PR DESCRIPTION
**Pull request by ahsan-storm**
Signed-off by: Muhammad Ahsan
username: ahsan-storm
email: ahsanmuhammad1@outlook.com

#### What this PR does / why we need it:
- Adds support for custom kubernetes resources
- Adds support for making configmap optional
- Adds support for annotations in daemonset
- Adds support for update strategy in daemonset
- Adds support for host network in daemonset
- Adds support for dns policy in daemonset
- Adds support for adding extra envs in daemonset

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by ahsan-storm** [comment]
@lachie83 kindly review this PR, and merge it in master

**Comment by ahsan-storm** [comment]
@monotek @lachie83 Kindly review this PR, and merge it in master

**Comment by monotek** [comment]
I'm currently waiting for somebody to merge: https://github.com/helm/charts/pull/9382
After that i'll have a look on this pr asap :-)

**Comment by monotek** [comment]
Please fix merge conflicts....

File: stable/fluentd-elasticsearch/templates/customResources.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{- if .Values.customResources -}}
```
Comment (monotek) [inline stable/fluentd-elasticsearch/templates/customResources.yaml:1]: What kind of custom resources are we talking about here?

Is it about Kubernetes CRDs? If not, the name is misleading. 
I would suggest "Additional resources" in this case.

File: stable/fluentd-elasticsearch/templates/daemonset.yaml [hunk 2]
```diff
@@ -10,7 +10,15 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     heritage: "{{ .Release.Service }}"
     release: "{{ .Release.Name }}"
+{{- if .Values.daemonset.annotations }}
+  annotations:
+{{ toYaml .Values.daemonset.annotations | indent 4 }}
+{{- end }}
 spec:
+{{- if .Values.daemonset.updateStrategy }}
+  updateStrategy:
```
Comment (monotek) [inline stable/fluentd-elasticsearch/templates/daemonset.yaml:19]: Imho we should set "RollingUpdate" as the default in the values.yaml so we can remove the if statement.

File: stable/fluentd-elasticsearch/templates/daemonset.yaml [hunk 3]
```diff
@@ -35,6 +43,12 @@ spec:
 {{ toYaml .Values.annotations | indent 8 }}
 {{- end }}
     spec:
+{{- if .Values.daemonset.hostNetwork }}
+      hostNetwork : {{ .Values.daemonset.hostNetwork }}
+{{- end}}
+{{- if .Values.daemonset.dnsPolicy }}
+      dnsPolicy: {{ .Values.daemonset.dnsPolicy }}
```
Comment (monotek) [inline stable/fluentd-elasticsearch/templates/daemonset.yaml:50]: Same as for updatestrategey. I'd like to see "Default" set in the values.yaml and the if statement removed.

File: stable/fluentd-elasticsearch/templates/daemonset.yaml [hunk 4]
```diff
@@ -35,6 +43,12 @@ spec:
 {{ toYaml .Values.annotations | indent 8 }}
 {{- end }}
     spec:
+{{- if .Values.daemonset.hostNetwork }}
+      hostNetwork : {{ .Values.daemonset.hostNetwork }}
```
Comment (monotek) [inline stable/fluentd-elasticsearch/templates/daemonset.yaml:47]: Why do you want to use hostNetwork?
Did you considered to use NodePort instead?

File: stable/fluentd-elasticsearch/values.yaml [hunk 5]
```diff
@@ -461,7 +461,25 @@ configMaps:
...
+# - apiVersion: custom-version
+#   kind: custom-kind
+#   any other property
+customResources: []
+daemonset:
+  annotations: {}
+  env: []
+  securityContext: {}
+  updateStrategy: {}
+  hostNetwork: true
```
Comment (monotek) [inline stable/fluentd-elasticsearch/values.yaml:481]: I don't want to see this set to "true" by default.
Using of hostNetwork is discouraged. See: https://kubernetes.io/docs/concepts/configuration/overview/#services

File: stable/fluentd-elasticsearch/values.yaml [hunk 6]
```diff
@@ -461,7 +461,25 @@ configMaps:
...
+  dnsPolicy: ClusterFirstWithHostNet
```
Comment (monotek) [inline stable/fluentd-elasticsearch/values.yaml:482]: Should be "Default" by default.

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by monotek** [comment]
@ahsan-storm 
Why do you have closed the pr?

**Comment by ahsan-storm** [comment]
@monotek I actually don't need these changes now

**Pull request closed without merging**